data/policies/parent_call.cas

```diff
@@ -33,9 +33,30 @@ resource xyz inherits abc {
 	}
 }
 
-// TODOs once we add the ability to cast to
-// non-inherited things
-//
-//domain asd {
-//	asd<abc>.read(this);
-//}
+domain asd {
```

Comment:
Can you also test this situation?

```
resource foo {
   fn some_func(domain source, resource some_resource) {
       allow(source, some_resource, file, read);
   }
}

@associate([tmp])
domain bar {
    foo.some_func(this, this.tmp);
}
```

Reply:
That will compile in cascade but not with secilc.
```
Failed to resolve qwe.tmp in call argument list at out.cil:201
Failed to resolve call statement at out.cil:201
Failed to resolve AST
Failed to compile cildb: -2
```

Reply:
Interesting.  CIL shouldn't be seeing "qwe.tmp" at all.  It should have been converted to "qwe-tmp".

Sounds like we may have a bug where we aren't converting associated resource names in a call argument context?  Can you see if it reproduces with a simple call situation without casting involved and make a second PR to fix it?

Actually, looking again, this is a non-casting situation.  So your casting code isn't even running here.  I'm not actually sure if triggering the case I was thinking of is possible in the current code.  Let me give it some more thought...

Reply:
bump on this:

> Sounds like we may have a bug where we aren't converting associated resource names in a call argument context? Can you see if it reproduces with a simple call situation without casting involved and make a second PR to fix it?

Reply:
https://github.com/dburgener/cascade/issues/133